### PR TITLE
28 add branch reports view attendance by day window

### LINKED ISSUE #28: Add branch reports view - attendance by day window
# REQUIREMENTS
- Add a list of employees by branch who attendance each day
- Add filters by day, position and name of employee
# DESIGN
image

### PR DESCRIPTION
**Pull request by Javi1499**
#Description
What did you implemented
- Creation of attendance employee table
-  Filters inputs was added to show results by date, position or name
- Gem faker added

Why did you implement this:
  - Attendance employee table was added to show information about check in/out of employees.
  - Filter feature was implemented to search specific information
  - Faker was use to create information to test functionality.
  
  ## Screenshots
Attendance by day

### CODE REVIEW

File: lib/tasks/data.rake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  role = rand(1..Role.count)
+  private_number = Faker::Number.number(digits: 5)
+  Employee.create(name: name, email: email, role_id: role, private_number: private_number, status: true , branch_id: branch)
+  }
+end
+
+task :roles => :environment do
+  15.times{
+  position = Faker::Job.position
+  Role.create(description:position)
```
Comment (YilKanda) [inline lib/tasks/data.rake]: Check your indentation

File: lib/tasks/data.rake [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+task :employee => :environment do
+  employee = rand(3)
+  100.times{
+  name = Faker::Name.name
+  email = Faker::Internet.email
+  branch = rand(1..Branch.count)
+  role = rand(1..Role.count)
+  private_number = Faker::Number.number(digits: 5)
+  Employee.create(name: name, email: email, role_id: role, private_number: private_number, status: true , branch_id: branch)
```
Comment (YilKanda) [inline lib/tasks/data.rake]: Check your indentation

File: app/views/branches/_filter.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<%= form_tag filter_path(@branch.id), method: :get, remote: true, class: 'filter' do %>
+  <div class="form-group w-75">
+    <label>Filter by day</label>
+    <%= date_field_tag :day, class:"form-date"%>
+  </div>
+  <div class="form-group w-75 my-4">
+    <label>Filter by role</label>
+    <%= select_tag :role_id, options_from_collection_for_select(Role.all, "id", "description"), prompt: "Select a role", class:"form-select"%>
```
Comment (YilKanda) [inline app/views/branches/_filter.erb]: It is not recommended that you leave model information in the view, I recommend that you call this information from a helper.

```suggestion
    <%= select_tag :role_id, options_from_collection_for_select(get_role, "id", "description"), prompt: "Select a role", class:"form-select"%>
```

In the helper:

```
# app/helpers/branch_helper.rb

module BranchHelper
  def get_role
    Role.all
  end
end
`

**Review by YilKanda — CHANGES_REQUESTED** [review]

File: app/views/branches/_filter.erb (hunk 3 above)
Comment (Javi1499) [reply app/views/branches/_filter.erb]: Thanks, I used it in my code .

File: app/views/branches/show.html.erb [hunk 4]
```diff
@@ -1,7 +1,33 @@
...
+  <div class="row">
+    <div class="col-7 d-flex justify-content-center">
+      <div class="w-100 table-attendance">
+        <%= render partial: 'attendance' %>
+      </div>
+    </div>
+    <div class="col-5 d-flex justify-content-center">
+      <div class="w-100">
+        <div id="filter">
+          <%= render :partial => 'branches/filter' %>
```
Comment (YilKanda) [inline app/views/branches/show.html.erb]:
```suggestion
          <%= render partial: 'branches/filter' %>
```

File: lib/tasks/data.rake [hunk 5]
```diff
@@ -0,0 +1,29 @@
+require 'faker'
+desc 'Populates the database with sample data'
+task :data => :environment do
```
Comment (YilKanda) [inline lib/tasks/data.rake]:
```suggestion
task data: :environment do
```

File: lib/tasks/data.rake [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  employee = rand(3)
+  150.times{|d|
+            (Employee.first.id .. Employee.last.id).each_with_index do  |e|
+            puts e
+            Record.create! employee_id: e, check_in: Time.now.midnight-d.day+8.hours, check_out: Time.now.midnight-d.day+16.hours
+            end}
+end
+
+task :employee => :environment do
```
Comment (YilKanda) [inline lib/tasks/data.rake]:
```suggestion
task employee: :environment do
```

File: lib/tasks/data.rake [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    name = Faker::Name.name
+    email = Faker::Internet.email
+    branch = rand(1..Branch.count)
+    role = rand(1..Role.count)
+    private_number = Faker::Number.number(digits: 5)
+    Employee.create(name: name, email: email, role_id: role, private_number: private_number, status: true , branch_id: branch)
+  }
+end
+
+task :roles => :environment do
```
Comment (YilKanda) [inline lib/tasks/data.rake]:
```suggestion
task roles: :environment do
```

**Review by YilKanda — DISMISSED** [review]
I'm just adding some hints regarding the latest Ruby code style trend.

File: app/views/branches/show.html.erb (hunk 4 above)
Comment (Javi1499) [reply app/views/branches/show.html.erb]: I'll make these changes, thanks

**Review by EliasUR — APPROVED** [review]
LGTM

**Review by femm34 — APPROVED** [review]
LGTM

**Review by germanruzca — APPROVED** [review]
LGTM

**Pull request merged**